packages/preview-middleware/src/base/config.ts

```diff
@@ -0,0 +1,396 @@
+export const PREVIEW_URL = {
+    client: {
+        url: '/preview/client',
+        local: join(__dirname, '../../dist/client'),
+        ns: 'open.ux.preview.client'
+    },
+    api: '/preview/api'
+} as const;
+
+/**
+ * Default theme
+ */
+export const DEFAULT_THEME = 'sap_horizon';
+
+/**
+ * Default path for mounting the local FLP.
+ */
+export const DEFAULT_PATH = '/test/flp.html';
+
+/**
+ * Default intent
+ */
+export const DEFAULT_INTENT = {
+    object: 'app',
+    action: 'preview'
+} as Readonly<Intent>;
+
+/**
+ * SAPUI5 delivered namespaces from https://ui5.sap.com/#/api/sap
+ */
+const UI5_LIBS = [
+    'sap.apf',
+    'sap.base',
+    'sap.chart',
+    'sap.collaboration',
+    'sap.f',
+    'sap.fe',
+    'sap.fileviewer',
+    'sap.gantt',
+    'sap.landvisz',
+    'sap.m',
+    'sap.ndc',
+    'sap.ovp',
+    'sap.rules',
+    'sap.suite',
+    'sap.tnt',
+    'sap.ui',
+    'sap.uiext',
+    'sap.ushell',
+    'sap.uxap',
+    'sap.viz',
+    'sap.webanalytics',
+    'sap.zen'
+] as const;
+
+/**
+ * Gets the UI5 libs dependencies from manifest.json.
+ *
+ * @param manifest application manifest
+ * @returns UI5 libs that should preloaded
+ */
+function getUI5Libs(manifest: Manifest): string {
+    const libs = manifest['sap.ui5']?.dependencies?.libs ?? {};
+    // add libs that should always be preloaded
+    libs['sap.m'] = {};
+    libs['sap.ui.core'] = {};
+    libs['sap.ushell'] = {};
+
+    return Object.keys(libs)
+        .filter((key) => {
+            return UI5_LIBS.some((substring) => {
+                return key === substring || key.startsWith(substring + '.');
+            });
+        })
+        .join(',');
+}
+
+/**
+ * Default configuration for the FLP.
+ *
+ * @param config partial configuration
+ * @returns a full configuration with default values
+ */
+export function getFlpConfigWithDefaults(config: Partial<FlpConfig> = {}) {
+    const flpConfig = {
+        path: config.path ?? DEFAULT_PATH,
+        intent: config.intent ?? DEFAULT_INTENT,
+        apps: config.apps ?? [],
+        libs: config.libs,
+        theme: config.theme,
+        init: config.init
+    };
+    if (!flpConfig.path.startsWith('/')) {
+        flpConfig.path = `/${flpConfig.path}`;
+    }
+    return flpConfig;
+}
+
+/**
+ * The developer mode is only supported for adaptation projects, therefore, notify the user if it is wrongly configured and then disable it.
+ *
+ * @param config configurations from the ui5.yaml
+ * @param logger logger instance
+ */
+export function sanitizeConfig(config: MiddlewareConfig, logger: ToolsLogger): void {
+    if (config.rta && config.adp === undefined) {
+        config.rta.editors = config.rta.editors.filter((editor) => {
+            if (editor.developerMode) {
+                logger.error('developerMode is ONLY supported for SAP UI5 adaptation projects.');
+                logger.warn(`developerMode for ${editor.path} disabled`);
+            }
+            return !editor.developerMode;
```

Comment:
Looks like the return value is not needed? function definition is also with void

Reply:
This is the nested filter function, and it requires a boolean as return value. Nevertheless, the functions should not be called for ADPs, therefore, I think, I incorrectly use it due to the refactoring. Let me check.

Reply:
OK